### PR TITLE
Slimmer OpenSSL

### PR DESCRIPTION
**Pull request by xnox**
Disable obsolete and deprecated build options:

 * disable DES and disable DSA. Both DES and DSA are obsoleted by NIST see:
    https://csrc.nist.gov/news/2023/nist-to-withdraw-sp-800-67-rev-2
    https://csrc.nist.gov/news/2023/nist-releases-fips-186-5-and-sp-800-186

 * disable engines support. Deprecated in 3.0 and no longer contain any interesting hardware acceleration. See: https://github.com/openssl/openssl/blob/master/README-ENGINES.md

 * drop fips packaging as it has moved elsewhere

 * dump config data after the build for ease of inspection

### CODE REVIEW

File: openssl.yaml [hunk 1]
```diff
@@ -60,7 +52,10 @@ pipeline:
          no-ssl3 \
          no-seed \
          no-weak-ssl-ciphers \
+         no-des \
+         no-dsa \
          -Wa,--noexecstack
+      perl configdata.pm --dump
```
Comment (vaikas) [inline openssl.yaml:58]: Just checking that this is not debug stuff left around since it wasn't there before.

**Review by vaikas — APPROVED** [review]

File: openssl.yaml [hunk 2]
```diff
@@ -12,14 +12,6 @@ package:
     runtime:
       - openssl-provider-legacy
 
-vars:
-  fips-enable: 'no-fips'
-
-options:
-  fips:
-    vars:
-      fips-enable: 'enable-fips'
```
Comment (imjasonh) [inline openssl.yaml:21]: This might be the last remaining usage of `options` in our repos, which means we might be able to remove this from Melange 👀

Comment (dlorenc) [reply openssl.yaml:21]: yay!

File: openssl.yaml (hunk 1 above)
Comment (xnox) [reply openssl.yaml:58]: Intentional commit and mentioned in the pull request description.

The output this produces allows inspection of all the things enabled and disabled. After configure is executed it even suggests to run that to see the configuration state.
Many keys are non declarative, and cascade enable/disable things in unexpected ways.

**Comment by xnox** [comment]
Well that didn't land well. https://github.com/wolfi-dev/os/issues/16698 on steps on how to land it better.

**Pull request merged**